src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -54,6 +54,7 @@ public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstruct
                 stores.Add(assignment);
             }
 
+            stores.Add(new BoundVoid(node.Syntax, node.Type));
```

Comment:
The context is that we need to build an expression with a void return type. As this expression is lowered into a sequence, the sequence automatically picks up its return type to be the type of it's last step. And there are checks all over the place that enforce that.
This is the approach we discussed with Vlad and Neal. We also considered tweaking the existing checks, or creating a VoidSequence, neither of which seems advantageous compared to `BoundVoid`. I'm happy to discuss options though.